src/main/java/pl/edu/agh/ki/io/api/FriendshipsApiController.java

```diff
@@ -0,0 +1,100 @@
+        }
+        return ResponseEntity.notFound().build();
+    }
+
+    @PostMapping("/api/users/{userid}/accept")
+    public ResponseEntity<?> acceptFriend(@AuthenticationPrincipal User user, @PathVariable("userid") Long requesterId) {
+        Optional<User> requester = this.userStorage.findUserById(requesterId);
+        if(requester.isPresent()) {
+            User addressee = ((UserPrincipal) this.userStorage.loadUserByUsername(user.getLogin())).getUser();
+            Friendship pendingFriendship = this.friendshipStorage.findByRequesterAndAddressee(requester.get(), addressee);
+
+            if(pendingFriendship != null) {
+                pendingFriendship.setStatus(1);
+                this.friendshipStorage.saveFriendship(pendingFriendship);
+
+                Friendship friendship = new Friendship(1, addressee, requester.get());
+                this.friendshipStorage.saveFriendship(friendship);
+                return new ResponseEntity<>(friendship, HttpStatus.OK);
+            }
+            return ResponseEntity.notFound().build();
+
+        }
+        return ResponseEntity.notFound().build();
+    }
+
+    @PostMapping("/api/users/{userid}/reject")
+    public ResponseEntity<?> rejectFriend(@AuthenticationPrincipal User user, @PathVariable("userid") Long requesterId) {
+        Optional<User> requester = this.userStorage.findUserById(requesterId);
+        if(requester.isPresent()) {
+            User addressee = ((UserPrincipal) this.userStorage.loadUserByUsername(user.getLogin())).getUser();
+            if(this.friendshipStorage.findByRequesterAndAddressee(requester.get(), addressee) != null){
+                this.friendshipStorage.deleteByRequesterAndAddressee(requester.get(), addressee);
+                return new ResponseEntity<>(HttpStatus.OK);
+            }
+            else return ResponseEntity.notFound().build();
+        }
+        return ResponseEntity.notFound().build();
+    }
+
+    @PostMapping("/api/users/{userid}/remove")
+    public ResponseEntity<Friendship> removeFriend(@AuthenticationPrincipal User user, @PathVariable("userid") Long userId) {
+        Optional<User> requester = this.userStorage.findUserById(userId);
+        if(requester.isPresent()) {
+            User addressee = ((UserPrincipal) this.userStorage.loadUserByUsername(user.getLogin())).getUser();
+            if(this.friendshipStorage.findByRequesterAndAddressee(requester.get(), addressee) != null &&
+                    this.friendshipStorage.findByRequesterAndAddressee(addressee, requester.get()) != null) {
+
+                this.friendshipStorage.deleteByRequesterAndAddressee(requester.get(), addressee);
+                this.friendshipStorage.deleteByRequesterAndAddressee(addressee, requester.get());
+
+                return new ResponseEntity<>(HttpStatus.OK);
+            }
+            return ResponseEntity.notFound().build();
+        }
+        return ResponseEntity.notFound().build();
+    }
+
```

Comment:
Jakiś endpoint na zablokowanie użytkownika?

Reply:
Na razie nie ma, w innym tasku się to zrobi